### ISSUE TITLE
Including files from withing sub directory with relative path is inconsistent/fails

### ISSUE THREAD

**Issue by Impact123** [alliedmodders/sourcepawn#245]
## Setup
I have a directory layout like this.

```
│   plugin.sp
└───includes
    │   extension.inc
    └───extension
            file.inc
```

`plugin.sp` includes `extension.inc` like this
```SourcePawn
#include "includes/extension"
```

`extension.inc` includes `file.inc` like this
```SourcePawn
#include "extension/file"
```

spcomp is in the `PATH` and i compile from the directory where `plugin.sp` is like this
```bash
spcomp plugin.sp
```
A test case that compiles on linux but fails on windows can be found here: [includebug.zip](https://github.com/alliedmodders/sourcepawn/files/2432797/includebug.zip).

The version i used were
**Windows**: `1.10.0.6336`
**Linux**: `1.10.0.6335`

## Description
The error i get is this
```
SourcePawn Compiler 1.10.0.6336
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2018 AlliedModders LLC

includes/extension.inc(1) : fatal error 183: cannot read from file: "extension/file"

Compilation aborted.
1 Error.
```

If i change the include inside `extension.inc` to
```SourcePawn
#include "includes/extension/file"
```
it works.

The problem is that this is inconsistent between windows and linux. If i make the change described here then it works on windows but not on linux.

## Expected solution
Ideally one would expect that the include paths are always relative to the file that includes them 
like in the attached [includebug.zip](https://github.com/alliedmodders/sourcepawn/files/2432797/includebug.zip).

Please let me know if you need more information.

**Comment by dordnung** [comment]
The problem is on this line: https://github.com/alliedmodders/sourcepawn/blob/master/compiler/lexer.cpp#L220
It tries to search relative to the last include file (which is **includes/extension** in the given example). But on the given line it retrieves the base path by finding the last directory separator. But there is no windows directory separator in **includes/extension**, which is the reason it works on linux but not on windows systems

**Comment by Impact123** [comment]
I don't feel confident enough in this to create a pull request, but here's some proof of concept patch that appears to fix this. I have only briefly tested this on windows
```diff
diff --git a/compiler/lexer.cpp b/compiler/lexer.cpp
index 28a5457..205b892 100644
--- a/compiler/lexer.cpp
+++ b/compiler/lexer.cpp
@@ -205,7 +205,16 @@ doinclude(int silent)
 
     i = 0;
     while (*lptr != c && *lptr != '\0' && i < sizeof name - 1) /* find the end of the string */
+    {
+        if (*lptr == '/' && DIRSEP_CHAR == '\\') {
+            name[i++] = DIRSEP_CHAR;
+            *lptr++;
+            continue;
+        }
+        
         name[i++] = *lptr++;
+    }
     while (i > 0 && name[i - 1] <= ' ')
         i--; /* strip trailing whitespace */
     assert(i < sizeof name);
```
Not sure if this would be the right approach here, but something tells me it might not be

**Comment by Fyren** [comment]
Should be fixed by #570.
